agent/api/api_client_test.go

```diff
@@ -0,0 +1,120 @@
+package api
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/aws/amazon-ecs-agent/agent/auth"
+	"github.com/aws/amazon-ecs-agent/agent/config"
+	svc "github.com/aws/amazon-ecs-agent/agent/ecs_client/ecs_autogenerated_client"
+)
+
+type MockAmazonEC2ContainerServiceV20141113Client struct {
+	svc.AmazonEC2ContainerServiceV20141113Client
+
+	lastRequest interface{}
+}
+
+func (mock *MockAmazonEC2ContainerServiceV20141113Client) SubmitContainerStateChange(req svc.SubmitContainerStateChangeRequest) (svc.SubmitContainerStateChangeResponse, error) {
+	mock.lastRequest = req
+
+	return svc.NewSubmitContainerStateChangeResponse(), nil
+}
+
+func NewMockClient() (ECSClient, *MockAmazonEC2ContainerServiceV20141113Client) {
```

Comment:
I'm okay with it being public because it's in _test anyways, so nothing can rely on it.